src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadataBuilder.cs

```diff
@@ -52,6 +55,8 @@ public class ClonedPackageSearchMetadata : IPackageSearchMetadata
 
             internal AsyncLazy<PackageDeprecationMetadata> LazyDeprecationFactory { get; set; }
             public async Task<PackageDeprecationMetadata> GetDeprecationMetadataAsync() => await (LazyDeprecationFactory ?? LazyNullDeprecationMetadata);
+            internal AsyncLazy<IEnumerable<PackageVulnerabilityMetadata>> LazyVulnerabilityFactory { get; set; }
+            public async Task<IEnumerable<PackageVulnerabilityMetadata>> GetVulnerabilityMetadataAsync() => await (LazyVulnerabilityFactory ?? LazyNullVulnerabilityMetadata);
```

Comment:
`LazyVulnerabilityFactory` never gets set anywhere, so will always be null, therefore `LazyNullVulnerabilityMetadata` will always be used.

Reply:
Thanks--I'm taking a look through this and seeing that it's when cloning search metadata objects that these methods get wrapped in lazys, like here: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadataBuilder.cs#L104
...and I missed adding this change for vulnerability metadata.